src/Bullets.ts

```diff
@@ -0,0 +1,95 @@
+export { FreezeBullet, Laser };
+
+import { constants } from './constants';
+
+// Bullet code
+//
+
+abstract class Bullet extends Phaser.Physics.Arcade.Sprite {
+  velX: number;
+  velY: number;
+  abstract speed: number;
```

Comment:
id probably just make this a readonly constructor-set value... seems constant in subclasses yea?

Reply:
I _think_ I resolved this as you intended, but let me know if I got it wrong.